namerd/iface/control-http/src/main/scala/io/buoyant/namerd/iface/HttpControlService.scala

```diff
@@ -284,44 +311,40 @@ class HttpControlService(storage: DtabStore, namers: Ns => NameInterpreter)
       }
     }
 
-  private[this] def renderTryTree(tryTree: Try[NameTree[Name.Bound]], closable: Closable) = tryTree match {
-    case Return(tree) =>
-      Buf.Utf8(tree.show + "\n")
-    case Throw(e) =>
-      closable.close()
-      Buf.Empty
-  }
-
-  private[this] def handleGetBind(ns: String, path: Path): Future[Response] = {
-    val act = getBind(ns, path)
-    streamingResp(act.values, renderTryTree)
-  }
+  private[this] def handleGetBind(ns: String, path: Path, req: Request): Future[Response] =
+    if (streaming(req)) {
+      streamingResp(getBind(ns, path).values) { tree =>
+        Buf.Utf8(tree.show + "\n")
+      }
+    } else {
+      getBind(ns, path).toFuture.map { tree =>
+        val rsp = Response()
+        rsp.content = Buf.Utf8(tree.show + "\n")
+        rsp
+      }
+    }
 
   private[this] val addrCacheMu = new {}
-  private[this] var addrCache: Map[(String, Path), Var[Addr]] = Map.empty
-  private[this] def getAddr(ns: String, path: Path): Var[Addr] = addrCacheMu.synchronized {
+  private[this] var addrCache: Map[(String, Path), Activity[Addr]] = Map.empty
+  private[this] def getAddr(ns: String, path: Path): Activity[Addr] = addrCacheMu.synchronized {
```

Comment:
The change in the return type is worth explaining.

Addr has a `Addr.Pending` value.  Previously, we flatMapped the `Activity[NameTree]` directly to a `Var[Addr]` by mapping `Activity.Pending` to `Addr.Pending`.  This leads to some questions about how this pending state should be displayed in non-streaming responses.  Typically, we use `Activity.toFuture` to get the first non-pending value from an Activity but that doesn't work here because `Addr.Pending` is, in some sense, a non-pending value.

Therefore, I changed this to flatMap the `Activity[NameTree]` into an `Activity[Addr]` so that it made sense to respond with the first non-pending value.